packages/cloud-functions/src/cico/moonpayWebhook.ts

```diff
@@ -0,0 +1,57 @@
+import crypto from 'crypto'
+import * as functions from 'firebase-functions'
+import { MOONPAY_WEBHOOK_KEY } from '../config'
+import { saveTxHashProvider } from '../firebase'
+import { Provider } from './Provider'
+
+const MOONPAY_SIGNATURE_HEADER = 'Moonpay-Signature-V2'
+
+function verifyMoonPaySignature(signatureHeader: string | undefined, body: string) {
+  if (!signatureHeader) {
+    return false
+  }
+
+  const [first, second] = signatureHeader.split(',')
+  const timestamp = first.split('=')[1]
+  const signature = second.split('=')[1]
+  const signedPayload = `${timestamp}.${body}`
+
+  const hmac = crypto.createHmac('sha256', MOONPAY_WEBHOOK_KEY)
+  hmac.write(signedPayload)
+  hmac.end()
+
+  const expectedSignature = Buffer.from(hmac.read().toString('hex'))
+  const signatureBuffer = Buffer.from(signature)
+  return Buffer.compare(signatureBuffer, expectedSignature) === 0
+}
+
+export const moonpayWebhook = functions.https.onRequest((request, response) => {
+  if (
+    verifyMoonPaySignature(request.header(MOONPAY_SIGNATURE_HEADER), JSON.stringify(request.body))
+  ) {
+    const {
+      data: { walletAddress, cryptoTransactionId },
+      type,
+    } = request.body
```

Comment:
Can you please create a Moonpay request body type so that someone who's new to the code knows what to expect?